common/app/common/ModelOrResult.scala

```diff
@@ -19,7 +19,34 @@ object ModelOrResult extends Results with Logging {
       .getOrElse(Right(NoCache(NotFound)))
 }
 
+object ContentRedirect {
```

Comment:
I think this is the most useful bit of this PR so will try and preserve it separately.